config/services/dataplane_v1beta1_openstackdataplaneservice_ovn.yaml

```diff
@@ -6,3 +6,16 @@ spec:
   playbook: osp.edpm.ovn
   configMaps:
   - ovncontroller-config
+  tlsCert:
```

Comment:
... i was not aware https://github.com/openstack-k8s-operators/dataplane-operator/pull/610 happened.

that is a very problematic change from my perspective.

we do not expect any customer to ever use the example nova service and we have even talked about removing it in the past.

With Kustomize we may be able to move too a model where the customer will tweak it via Customize but exposing this level of configuratblity without have a type filed or similar that we can key off is going to result in customer bugs...

I'm not going to propose a revert now but this is not a design i would have been ok with if i had had time to review it in january.

Reply:
@SeanMooney I'm not sure which part you're referring to, but there was always a need to specify which services would need a certificate and which one's would not.  In the past, this was a parameter tlsCertEnabled or some such in the spec.

The refactor resulted from @slagle et al insistence that the dataplane operator not have dataplane service specific logic in it.  ie. no case statements with service specific logic depending on the service.  Instead all services are treated equally and any service specific logic is triggered off of service spec parameters.

If you'd prefer something different, then lets have the debate!  @slagle ^^

Reply:
yes i do think we need something a lot different form the current state.

ideally before ga.

when we had the boolean the intent was that by default the value was taken form the node set and then you could opt out of the node set value if you defined it in the service

at least that was my understanding so we would not normally expect that an end user creating a custom nova dataplane service would have to set this.

with he current structure specifying the networks/keyusages  and the fact that we never expect the nova dataplane service to be used by customer this opens a vector to misconfigure the tls cert generation that did not exist before.

where my head is right now is we should likely move away form autocatratin some or all of the data plane services and instead auto create a set of dataplane service schma/type validation

i.e. we should have a data driven way to specific that for a given type of DataplaneSerivce the following fields are required/optional and we should be able to express that with conditionality. i.e. the following is required if tls is enabled but not otherwise.

ideally that could be used by both defaultign and validating webhooks too when the user creates a concrete dataplane service with the type filed set to the name of the schema cr.

I'm borderline burnt out at the moment so i dint really have the mental energy to express this in detail right now but if you or @slagle want to chat about this at some point then lets have an informal call. this is post beta and likely post ga work but it ties into https://issues.redhat.com/browse/OSPRH-354 albeit a little tangentlely.

Reply:
to be clear for GA we can mitigate this using kustomize to extend the default samples in our docs but long term i would like to have a better solution for validating the dataplane service with knowledge of the service requirements rather then just an entirely
generic interface.

Reply:
this PR isn't really the place for this discussion but I agree the dataplane TLS config currently places a lot of responsibility on the human operator. On the control plane it's just toggling a bool and the default will be true once everything is in place. Ideally on the dataplane side would "just work" out of the box like this too

Reply:
...but right now I think the priority is to get the remaining PRs landed and to get TLS enabled for CI/QE

Reply:
yep it is 
we just need to be carful in our docs to make sure that we call out the importance of the tls parmaters

Reply:
lets resolve this for now and loop back to it later